src/usePerpsDeposit.ts

```diff
@@ -0,0 +1,38 @@
+import { useMutation } from '@tanstack/react-query';
+import { ethers } from 'ethers';
+import { useSynthetix } from './useSynthetix';
+
+export function usePerpsDeposit({
```

Comment:
looking at this function it is not a perps collateral deposit. It looks like WETH deposit (ETH -> WETH).

This needs to be fully renamed to "useWethDeposit", it has no relation to perps


```suggestion
export function useWethDeposit({
```

The next issue is `tokenAddress` argument. As WETH address is predefined per chain and can be taken from `viem` chain definitions. So this argument should not be here as well as it is not an optional thing

Reply:
Renamed to `useWethDeposit`.

Added `useImportWethContract` hook, with `importWeth` function which return **WETH** contract address and abi. 

There is also a **TODO: define correct WETH contract addresses for this chains**